src/dvc_data/hashfile/gc.py

```diff
@@ -38,7 +38,8 @@ def _is_dir_hash(_hash):
 
     for paths in (dir_paths, file_paths):
         if paths:
-            removed = True
-            odb.fs.remove(paths)
+            removed += len(paths)
+            if not dry:
+                odb.fs.remove(paths)
```

Comment:
Maybe we could put some logging here in debug mode (or even info if you passed `dry`?
Or is the list of things to be removed logged elsewhere?

Reply:
Didn't track down where, but it looks like it is already logged:

```console
$ dvc gc -v --all-commits --all-branches --all-tags
2023-09-15 10:54:20,087 DEBUG: v3.19.1.dev20+gbf2af8639.d20230915, CPython 3.11.5 on macOS-13.5-arm64-arm-64bit
2023-09-15 10:54:20,087 DEBUG: command: /Users/dave/micromamba/envs/dvc-test/bin/dvc gc -v --all-commits --all-branches --all-tags
2023-09-15 10:54:20,288 WARNING: This will remove all cache except items used in the workspace and all git commits of the current repo.
Are you sure you want to proceed? [y/n]: y
2023-09-15 10:54:22,624 DEBUG: Estimated remote size: 256 files
2023-09-15 10:54:22,625 DEBUG: Removing '/Users/dave/repo/.dvc/cache/files/md5/91/52ab3f172b3e23fd6b1a3ab0e1d150'
Removed 1 objects from repo cache.
2023-09-15 10:54:22,625 DEBUG: Estimated remote size: 256 files
No unused 'local' cache to remove.
2023-09-15 10:54:22,625 DEBUG: Estimated remote size: 256 files
2023-09-15 10:54:22,625 DEBUG: '/Users/dave/repo/.dvc/cache/files/md5/5f/b7ba7e8447a836e774b66155f5776a' doesn't look like a cache file, skipping
No unused 'legacy' cache to remove.
```